frontend/src/components/main-page/graph-page/dependency-graph-view.tsx

```diff
@@ -53,41 +59,112 @@ export const DependencyGraph: React.FC = () => {
       },
     },
   ];
+  let cyInstance: cytoscape.Core | undefined;
+
+  const onDownload = async (): Promise<void> => {
+    if (cyInstance === undefined) {
+      return;
+    }
+    const pngBlob = await cyInstance.png({
+      output: 'blob-promise',
+    });
+
+    const fileName = 'msadoc-dependency-graph.png';
+    const downloadLink = document.createElement('a');
+    downloadLink.href = URL.createObjectURL(pngBlob);
+    downloadLink.download = fileName;
+    downloadLink.click();
+  };
```

Comment:
I would move this (and the entire rest of the logic) into the Controller. And maybe call this function "performDownload" or "initiateDownload" or "downloadAsPNG" or sth like that.

Reply:
The issue is that I need the reference to the cytoscape instance.. Should I put that into the controller/state as well? We get the reference by a callback:

```html
<CytoscapeComponent
    ...
    cy={(cy): void => {
      // save the reference cy in the state?
    }}
/>
```

Reply:
Tried it saving in the state: endless loop.

Rendered `CytoscapeComponent` sets the state, leading to rendering `CytoscapeComponent` and so forth :)

Reply:
Awesome, thx!